File: src/main/java/dirtymaster/fgpt/config/CompletionConfiguration.java
package dirtymaster.fgpt.config;

import lombok.Data;

import java.math.BigDecimal;

@Data
public class CompletionConfiguration {
    private Integer maxTokens;// = Integer.MAX_VALUE;
    private BigDecimal temperature = BigDecimal.valueOf(0.7);
    private Integer topK;
    private BigDecimal topP;
}

File: src/main/java/dirtymaster/fgpt/config/CompletionConfiguration.kt
package dirtymaster.fgpt.config

import java.math.BigDecimal

class CompletionConfiguration {
    val maxTokens: Int = Int.MAX_VALUE
    val temperature: BigDecimal = BigDecimal.valueOf(0.7)
    val topK: Int? = null
    val topP: BigDecimal? = null
}
